Device-omiauth/app/models/user.rb

```diff
@@ -2,5 +2,13 @@ class User < ActiveRecord::Base
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :trackable, :validatable
+    :omniauthable, omniauth_providers: [:google_oauth2]
+
+  def self.find_for_google_oauth2(access_token, signed_in_resource = nil)
+    email = access_token.info['email']
+    User.where(email: email).first_or_create do |user|
+      user.password = user.password_confirmation = Devise.friendly_token
+      user.save
```

Comment:
You need to set up the provider: provider and uid to the user. Those values should in the access_token.info hash